### PR TITLE
[fix](jdbc catalog) Fix type recognition error when using `query` tvf to query doris

### PR DESCRIPTION
**Pull request by zy-kkk**
Using string to match Doris type will not work with query tvf, so use field matching instead

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java [hunk 1]
```diff
@@ -374,11 +377,11 @@ private Type dorisTypeToDoris(String type) {
...
-            case "VARCHAR": {
-                int length = Integer.parseInt(upperType.substring(openParen + 1, upperType.length() - 1));
-                return baseType.equals("CHAR")
-                    ? ScalarType.createCharType(length) : ScalarType.createVarcharType(length);
-            }
+                ScalarType charType = ScalarType.createType(PrimitiveType.CHAR);
+                charType.setLength(fieldSchema.getColumnSize().orElse(0));
+                return charType;
+            case "VARCHAR":
+                return ScalarType.createVarcharType(fieldSchema.getColumnSize().orElse(0));
```
Comment (0xderek) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java:384]: Curious to know why do we set length to 0 when column size is not set?

Comment (zy-kkk) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java:384]: Thanks for finding this problem. The processing here is not appropriate, so I changed the behavior here, refer to https://github.com/apache/doris/pull/41243

**Pull request merged**